src/router/exchange_router.cairo

```diff
@@ -285,20 +285,20 @@ mod ExchangeRouter {
 
         fn create_deposit(ref self: ContractState, params: CreateDepositParams) -> felt252 {
             let data_store = self.data_store.read();
-            global_reentrancy_guard::non_reentrant_before(data_store);
+            // global_reentrancy_guard::non_reentrant_before(data_store);
 
             let account = get_caller_address();
 
             let key = self.deposit_handler.read().create_deposit(account, params);
 
-            global_reentrancy_guard::non_reentrant_after(data_store);
+            // global_reentrancy_guard::non_reentrant_after(data_store);
 
             key
         }
 
         fn cancel_deposit(ref self: ContractState, key: felt252) {
             let data_store = self.data_store.read();
-            global_reentrancy_guard::non_reentrant_before(data_store);
+            // global_reentrancy_guard::non_reentrant_before(data_store);
```

Comment:
The` global_reentrancy_guard::non_reentrant_before` function is also called in the `cancel_deposit` function.
So I think there's one call too many, please, can you check if it's really this call that should be removed and not the one from the `cancel_deposit` function?